packages/gateway/tests/integration/connection.spec.ts

```diff
@@ -162,7 +161,7 @@ describe('gateway', () => {
     const timeout = setTimeout(() => {
       throw new Error('Not heartbeat in time')
     }, 1017)
```

Comment:
Why is the duration so specific??? Like I would expect it to be 1000ms or something like that

Reply:
I have no clue why is it this much specific

Reply:
At first glance, it looks like something you would get as a factor of some equation, though I couldn't guess where it would come from either.

Reply:
Seems to be a value to account for the jitter, but the uWS uses the same value as hearbeat_interval so i don't understand what is going on this this value

Reply:
The value turns out to be a value to make `Shard.calculateSafeRequests` return something that isn't 0. `calculateSafeRequests` is there for determinate how many messages we can send before a shard gets its ratelimit reset & send enough heartbeats for it to not get closed by discord. The previous value (1017) is the lowest value for which the calculation `120 - (Math.ceil(60_000 / <heartbeat_interval>) * 2)`[^calc] returns 2. I have changed the value to 1050 to make it seem a little less odd and add a comment to explain what is that value and to make aware anyone editing the test that it has the `1017` constrain as a minimum value.

[^calc]: I have inlined some constants, both 120 & 60_000 are from the Shard class.

Reply:
For future references on why this number was chosen

![image](https://github.com/discordeno/discordeno/assets/87189679/03e2307d-ac34-49cb-a0b4-11cf39254ddf)